lib/cdo/secrets_config.rb

```diff
@@ -63,6 +63,7 @@ def to_s
 
       # Resolve the secret referenced by this object.
       def lookup(secrets_manager)
+        raise "Secret path is nil or empty. {secret_key: #{secret_key}}" if key.nil? || key.empty?
```

Comment:
Our secrets-config logic is complex. There's a risk our unit tests do not adequately exercise all of the different use cases to ensure this change is safe. Does the `test_stack_secret_tag_falls_back_to_regular_secret` [unit test](https://github.com/code-dot-org/code-dot-org/blob/6027b1d0333a54fa216ff43ab616b7328cfdda89/lib/test/cdo/test_secrets_config.rb#L92-L100) verify that it's safe to raise when the secret path is nil or empty?

There might be a flaw in [how we stub secret lookups](https://github.com/code-dot-org/code-dot-org/blob/6027b1d0333a54fa216ff43ab616b7328cfdda89/lib/test/cdo/test_secrets_config.rb#L93)  in that test that is hiding a regression in this change.